app/controllers/ops_controller/settings/label_tag_mapping.rb

```diff
@@ -52,9 +52,16 @@ def label_tag_mapping_edit
 
   def entity_ui_name_or_all(entity)
     if entity
-      entity = entity.split('::').last
-      entity = 'VmOrTemplate' if entity == 'Image'
-      ui_lookup(:model => entity)
+      provider, model = entity.split('::')
+
+      case model
+      when 'Vm'
+        model = "ManageIQ::Providers::#{provider}::CloudManager::Vm"
+      when 'Image'
+        model = "ManageIQ::Providers::#{provider}::CloudManager::Template"
```

Comment:
clever.

MAPPABLE_ENTITIES const is increasingly coupled to UI logic, would be nice to move it here from core repo (or maybe you have better plans with provider column).